src/main/java/com/google/gwt/site/webapp/client/GWTProjectEntryPoint.java

```diff
@@ -195,118 +245,130 @@ public void f() {
+    return !modifiers && !middle && !right;
   }
 
-  private boolean shouldEnhanceLink(GQuery link) {
+  private boolean shouldEnhanceLink(Element link) {
     return
         // Enhance only local links
-        isSameOriginRexp.test(link.attr("href")) &&
+        isSameOriginRexp.test(link.getAttribute("href")) &&
         // Do not load links that are marked as full page reload
-        !Boolean.parseBoolean(link.attr("data-full-load"));
+        !Boolean.parseBoolean(link.getAttribute("data-full-load"));
   }
 
   private void enhanceLinks() {
     // Replace relative paths in anchors by absolute ones
     // exclude all anchors in the conHighlight and collapse menutent area.
     // TODO could be done on server side
-    $("a").not($("#content a")).each(new Function() {
-      @Override
-      public void f(Element e) {
-        GQuery link = $(e);
-        if (shouldEnhanceLink(link)) {
-          // No need to make complicated things for computing
-          // the absolute path: anchor.pathname is the way
-          link.attr("href", link.prop("pathname"));
-        }
-      }
+
+    forEach("a", el -> {
+       if (el.closest("#content") != null) {
+         return;
+       }
+       if (shouldEnhanceLink(el)) {
+         // No need to make complicated things for computing
+         // the absolute path: anchor.pathname is the way
+         HTMLAnchorElement link = (HTMLAnchorElement) el;
+         link.href = link.pathname;
+       }
     });
 
     // We add a span with the +/- icon so as the click area is well defined
     // this span is not rendered in server side because it is only needed
     // for enhancing the page.
-    GQuery parentItems = $("#submenu ul > li li").has("ul").prepend("<span/>");
-
-    $("#submenu").children("span, a[href=\"#\"]").unbind("click");
+    List<HTMLElement> parentItems = new ArrayList<>();
+    forEach("#submenu ul > li li", el -> {
+      if (el.querySelector("ul") != null) {
+        parentItems.add(el);
+        el.parentElement.insertBefore(document.createElement("span"), el);
+      }
+    });
+    String spanOrLocalLink = "span,a[href^='#']";
+    forEachChild(querySelector("#submenu"), spanOrLocalLink, el -> {
+      EventListener clickListener = (EventListener) Js.asPropertyMap(el).get("clickListener");
+      el.removeEventListener("click", clickListener);
+    });
 
     // Toggle the branch when clicking on the arrow or anchor without content
-    $(parentItems).children("span, a[href=\"#\"]").on("click", new Function() {
-      @Override
-      public boolean f(Event e) {
-        toggleMenu($(e).parent());
-        return false;
+    parentItems.forEach(item -> {
+      forEachChild(item, spanOrLocalLink, element -> {
+        EventListener eventListener = (evt) ->
+            toggleMenu((HTMLElement) ((Element) evt.target).parentElement);
+        element.addEventListener("click", eventListener);
+        Js.asPropertyMap(element).set("clickListener", eventListener);
```

Comment:
Here, above, it might be nice to use JsWeakMap from elemental2-core - effectively does the same thing (non-iterable mapping, won't leak values when keys are removed) but without the "just treat arbitrary objects as maps" irritation.